nodebuilder/header/service.go

```diff
@@ -51,6 +52,33 @@ func (s *Service) GetVerifiedRangeByHeight(
 }
 
 func (s *Service) GetByHeight(ctx context.Context, height uint64) (*header.ExtendedHeader, error) {
+	// TODO(vgonkivs): remove after https://github.com/celestiaorg/go-header/issues/32 will be
+	// implemented
+	head, err := s.store.Head(ctx)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I was wrong about checking local first. We should check against the network Head first to catch the header from future errors. If we do local first, even when the head is from the future, we will get syncing still in progress, while it's not.